### PR TITLE
Complete upload functionality validation and testing

### PR DESCRIPTION
**Pull request by MH4GF**
## Summary

This PR completes the implementation and testing of the "Start from Upload" functionality, resolving the form validation issues and improving the user experience.

### 🐛 Bug Fixes
- **Fix drag & drop file handling** (`aded201c`) - Files dropped via drag & drop are now properly synced with the HTML input element using DataTransfer API
- **Fix form field validation** (`2342df6c`) - Changed textarea name from 'message' to 'initialMessage' to match the validation schema
- **Improve invalid file validation** (`3f8bf00f`) - Submit button is now properly disabled when invalid file formats are selected

### 🧪 Testing & Refactoring  
- **Add comprehensive validation utility** (`c611a893`) - Created `calculateHasContent` function with 10 test cases covering all validation scenarios
- **Refactor component validation** (`cf651cd1`) - Replaced inline validation logic with the new testable utility function

## New Validation Rules

The upload form now supports flexible submission modes:

| Scenario | Result |
|----------|--------|
| ✅ Valid file (.sql, .rb, .prisma, .json) + Message | Submittable |
| ✅ Valid file + Attachments (no message) | Submittable |  
| ✅ Message only (no file) | Submittable |
| ❌ Valid file only (no content) | Not submittable |
| ❌ Invalid file + Message | Not submittable |

## Test Coverage

- **10 comprehensive test cases** for validation logic
- **TDD approach** with pure, testable functions
- **Edge cases covered**: whitespace handling, empty forms, invalid files

## Technical Details

- **DataTransfer API** ensures drag & drop files are properly included in form submission
- **Form validation** now correctly matches backend schema expectations
- **Type-safe validation** with comprehensive TypeScript support
- **Maintainable code** with separated concerns and testable utilities

## Resolves

- resolve: https://github.com/route06/liam-internal/issues/5167

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

* **New Features**
  * Improved file input synchronization in the upload session form for a more consistent user experience.
  * Enhanced form validation to better determine when the form has sufficient content for submission.

* **Bug Fixes**
  * Addressed issues where the file input's internal state could become out of sync with the displayed selection.

* **Tests**
  * Added comprehensive tests to ensure accurate detection of form content presence.

### CODE REVIEW

File: frontend/apps/app/features/sessions/components/UploadSessionForm/UploadSessionFormPresenter/UploadSessionFormPresenter.tsx [hunk 1]
```diff
@@ -69,8 +79,12 @@ export const UploadSessionFormPresenter: FC<Props> = ({
...
 
   // Calculate hasContent for Enter key submission
-  const hasContent =
-    !!selectedFile || textContent.trim().length > 0 || attachments.length > 0
+  const hasContent = calculateHasContent({
+    selectedFile,
+    schemaStatus,
+    textContent,
+    attachments,
+  })
```
Comment (NoritakaIkeda) [inline frontend/apps/app/features/sessions/components/UploadSessionForm/UploadSessionFormPresenter/UploadSessionFormPresenter.tsx:87]: I think we can now upload more safely!

**Review by NoritakaIkeda — APPROVED** [review]
LGTM!
